.buildkite/scripts/run_autoformat.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# -*- encoding: utf-8
+"""
+This script does autoformatting in Buildkite on pull requests.
+
+In particular, it runs the 'make format' task, and if there are any changes,
+it pushes a new commit to your pull request and aborts the current build.
+"""
+
+import os
+import subprocess
+import sys
+
+
+def branch_name():
+    """Return the name of the branch under test."""
+    return os.environ["BUILDKITE_BRANCH"]
```

Comment:
minor: I wonder if it's worth pulling stuff like this into a separate file, say `provider.py`?

Then this file would have `from provider import branch_name`, and all the provider-specific code is in one place (and can be swapped out again later).